File: app/src/main/java/com/chad/baserecyclerviewadapterhelper/EmptyViewUseActivity.java
package com.chad.baserecyclerviewadapterhelper;

import android.os.Bundle;
import android.os.Handler;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;

import com.chad.baserecyclerviewadapterhelper.adapter.QuickAdapter;
import com.chad.baserecyclerviewadapterhelper.base.BaseActivity;
import com.chad.baserecyclerviewadapterhelper.data.DataServer;

public class EmptyViewUseActivity extends BaseActivity implements View.OnClickListener {
    private RecyclerView mRecyclerView;
    private QuickAdapter mQuickAdapter;
    private View notDataView;
    private View errorView;
    private boolean mError = true;
    private boolean mNoData = true;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setBackBtn();
        setTitle("EmptyView Use");
        setContentView(R.layout.activity_empty_view_use);
        findViewById(R.id.btn_reset).setOnClickListener(this);
        mRecyclerView = (RecyclerView) findViewById(R.id.rv_list);
        mRecyclerView.setHasFixedSize(true);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));

        notDataView = getLayoutInflater().inflate(R.layout.empty_view, (ViewGroup) mRecyclerView.getParent(), false);
        notDataView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onRefresh();
            }
        });
        errorView = getLayoutInflater().inflate(R.layout.error_view, (ViewGroup) mRecyclerView.getParent(), false);
        errorView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onRefresh();
            }
        });
        initAdapter();
        onRefresh();
    }

    private void initAdapter() {
        mQuickAdapter = new QuickAdapter(0);
        mRecyclerView.setAdapter(mQuickAdapter);
    }

    @Override
    public void onClick(View v) {
        mError = true;
        mNoData = true;
        mQuickAdapter.setNewData(null);
        onRefresh();
    }



    private void onRefresh() {
        mQuickAdapter.setEmptyView(R.layout.loading_view, (ViewGroup) mRecyclerView.getParent());
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                if (mError) {
                    mQuickAdapter.setEmptyView(errorView);
                    mError = false;
                } else {
                    if (mNoData) {
                        mQuickAdapter.setEmptyView(notDataView);
                        mNoData = false;
                    } else {
                        mQuickAdapter.setNewData(DataServer.getSampleData(10));
                    }
                }
            }
        }, 1000);
    }
}

File: kotlinapp/src/main/java/com/allen/kotlinapp/EmptyViewUseActivity.kt
package com.allen.kotlinapp

import android.os.Bundle
import android.os.Handler
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import com.allen.kotlinapp.adapter.QuickAdapter
import com.allen.kotlinapp.base.BaseActivity
import com.allen.kotlinapp.data.DataServer

/**
 * 文 件 名: EmptyViewUseActivity
 * 创 建 人: Allen
 * 创建日期: 2017/6/15 10:34
 * 修改时间：
 * 修改备注：
 */
class EmptyViewUseActivity : BaseActivity(), View.OnClickListener {
    private var mRecyclerView: RecyclerView? = null
    private var mQuickAdapter: QuickAdapter? = null
    private var notDataView: View? = null
    private var errorView: View? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setBackBtn()
        setTitle("EmptyView Use")
        setContentView(R.layout.activity_empty_view_use)
        findViewById(R.id.btn_reset).setOnClickListener(this)
        mRecyclerView = findViewById(R.id.rv_list) as RecyclerView
        mRecyclerView?.setHasFixedSize(true)
        mRecyclerView?.layoutManager = LinearLayoutManager(this)

        notDataView = layoutInflater.inflate(R.layout.empty_view, mRecyclerView?.parent as ViewGroup, false)
        notDataView?.setOnClickListener { onRefresh() }
        errorView = layoutInflater.inflate(R.layout.error_view, mRecyclerView?.parent as ViewGroup, false)
        errorView?.setOnClickListener { onRefresh() }
        initAdapter()
        onRefresh()
    }

    private fun initAdapter() {
        mQuickAdapter = QuickAdapter(0)
        mRecyclerView?.adapter = mQuickAdapter
    }

    override fun onClick(v: View) {
        mError = true
        mNoData = true
        mQuickAdapter?.setNewData(null)
        onRefresh()
    }

    private var mError = true
    private var mNoData = true

    private fun onRefresh() {
        mQuickAdapter?.setEmptyView(R.layout.loading_view, mRecyclerView?.parent as ViewGroup)
        Handler().postDelayed({
            if (mError) {
                mQuickAdapter?.emptyView = errorView
                mError = false
            } else {
                if (mNoData) {
                    mQuickAdapter?.emptyView = notDataView
                    mNoData = false
                } else {
                    mQuickAdapter?.setNewData(DataServer.getSampleData(10))
                }
            }
        }, 1000)
    }
}

